crates/autopilot/src/domain/settlement/competition.rs

```diff
@@ -0,0 +1,18 @@
+//! Auction data related to the specific settlement.
+
+use {
+    crate::domain::{self, eth},
+    std::collections::HashMap,
+};
+
+#[derive(Debug)]
+pub struct Auction {
+    pub id: domain::auction::Id,
+    /// Auction external prices
+    pub prices: domain::auction::Prices,
+    /// Settlement should appear onchain before this block.
+    pub deadline: eth::BlockNo,
```

Comment:
Is this really something specific to the auction? 
To me an auction can exist without a solution. However, looking at `get_auction` we can not recover this type unless there is a solution. Therefore, would it make more sense to have this field on the solution (or make it recoverable even if there was no solution?

I guess this would bring us back to creating a `settlement::Solution` type which differs from `competition::Solution` in
1. Not having a solution id
2. Having a deadline
3. (Possibly not needing the prices)

I might actually prefer this setup

<img src="https://media2.giphy.com/media/3ohA2ZD9EkeK2AyfdK/giphy.gif"/>